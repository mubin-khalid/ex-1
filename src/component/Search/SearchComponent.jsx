import React from 'react';
import {
  Layout, Affix, Input, Row, Col,
} from 'antd';
import PropTypes from 'prop-types';

const {
  Header,
} = Layout;
const SearchComponent = ({ handleSearchTerm }) => (
  <Affix>
    <Row
      justify="center"
      gutter={10}
      align="middle"
      className="search"
    >
      <Col
        span={12}
      >
        <Input placeholder="Search" onChange={handleSearchTerm} />
      </Col>
    </Row>
  </Affix>
);

Header.propTypes = {
  handleSearchTerm: PropTypes.func,
};
export default SearchComponent;
