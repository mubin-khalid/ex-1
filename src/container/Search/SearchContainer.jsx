import React from 'react';
import { useDispatch } from "react-redux";
import { searchUsers } from '../../redux/actions/userActions';
import HeaderComponent from '../../component/Search/SearchComponent';

const SearchContainer = () => {
  const dispatch = useDispatch();
  const handleSearchTerm = (e) => {
    dispatch(searchUsers(e.target.value))
  };
  return <HeaderComponent handleSearchTerm={handleSearchTerm} />;
};

export default SearchContainer;
