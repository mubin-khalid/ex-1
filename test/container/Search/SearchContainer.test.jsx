import React from "react";
import store from "../../../src/redux/store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import Search from "../../../src/container/Search/SearchContainer";

it("renders correctly", () => {
  const tree = renderer
  .create(
    <Provider store={store}>
      <Search />
    </Provider>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});