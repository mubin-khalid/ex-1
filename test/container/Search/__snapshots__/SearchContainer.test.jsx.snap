// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className=""
  >
    <div
      className="ant-row ant-row-center ant-row-middle search"
      style={
        Object {
          "marginLeft": -5,
          "marginRight": -5,
        }
      }
    >
      <div
        className="ant-col ant-col-12"
        style={
          Object {
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        <input
          className="ant-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
